xaitk_saliency/utils/masking.py

```diff
@@ -28,3 +29,36 @@ def generate_masked_images(
         masked_img = np.multiply(mask, img, casting='unsafe')
         masked_imgs.append(PIL.Image.fromarray(np.uint8(masked_img)))
     return masked_imgs
+
+
+def weight_regions_by_scalar(
```

Comment:
Similarly, there may be a vectorized form of this function to account for multiple classes as the same time.

Reply:
Done, I vectorized this to perform a weighted average of all classes and regions of the image.